nemo/collections/tts/modules/transformer_dec24.py

```
@@ -0,0 +1,691 @@
+from torch import nn
+from torch.cuda import amp
+from torch.nn.utils.rnn import pad_sequence
+
+
+# alibi helper functions (start)
+def get_relative_positions(seq_len: int, device, symmetric: bool = True) -> torch.tensor:
+    x = torch.arange(seq_len)[None, :]
+    y = torch.arange(seq_len)[:, None]
+    rel_pos = (x - y).to(device)
+    if symmetric:
+        rel_pos = -rel_pos.abs()
+    return rel_pos
+
+
+def get_alibi_slope(num_heads):
+    x = (2**8) ** (1 / num_heads)
+    return torch.tensor([1 / x ** (i + 1) for i in range(num_heads)]).unsqueeze(-1).unsqueeze(-1)
+
+
+# alibi helper functions (end)
+
+
+class ConvNorm(torch.nn.Module):
+    def __init__(
+        self,
+        in_channels,
+        out_channels,
+        kernel_size=1,
+        stride=1,
+        padding=None,
+        dilation=1,
+        bias=True,
+        w_init_gain='gpt2',
+        is_causal=False,
+        norm_name='',
+    ):
+        super(ConvNorm, self).__init__()
+
+        padding = 0 if is_causal else padding
+        if padding is None:
+            assert kernel_size % 2 == 1
+            padding = int(dilation * (kernel_size - 1) / 2)
+
+        self.is_causal = is_causal
+        self.kernel_size = kernel_size
+        self.dilation = dilation
+
+        self.conv = torch.nn.Conv1d(
+            in_channels,
+            out_channels,
+            kernel_size=kernel_size,
+            stride=stride,
+            padding=padding,
+            dilation=dilation,
+            bias=bias,
+        )
+
+        if w_init_gain == 'gpt2':
+            torch.nn.init.normal_(self.conv.weight, mean=0.0, std=0.02)
+        else:
+            torch.nn.init.xavier_uniform_(self.conv.weight, gain=torch.nn.init.calculate_gain(w_init_gain))
+
+        self.norm_name = norm_name
+        if self.norm_name == 'weightnorm':
+            self.conv = nn.utils.weight_norm(self.conv)
+        elif self.norm_name == 'spectralnorm':
+            self.conv = nn.utils.spectral_norm(self.conv)
```

If this were ConvNorm in a common module file, then I having weight and spectral norm might make sense. But if it is only used inside this Transformer class, then I think we can safely remove these.

If we did keep this config, then it should throw an error if `norm_name` is an unknown value.

---

This has been removed